### PR TITLE
Implementation of Issue #339 [Feature Request][ATB] Prisoner ransom instead of bonus

### PR DESCRIPTION
**Pull request by NickAragua**
Changes are as follows:
- removes 50k "prisoner capture bonus"
- on personnel tab, if all selected personnel are prisoners, "ransom" menu item shows up. Click to display confirmation dialog. Once 'ok' is clicked, it makes a daily log entry and adds the funds to your bank account, as well as removing all the personnel involved.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 1]
```diff
@@ -165,6 +166,9 @@
     public static final int DESIG_NUM     = 14;
     
     public static final String LOGTYPE_MEDICAL = "med";
+    
+    private static final HashMap<Integer, Integer> MECHWARRIOR_AERO_RANSOM_VALUES; 
+    private static final HashMap<Integer, Integer> OTHER_RANSOM_VALUES;
```
Comment (dericpage) [inline MekHQ/src/mekhq/campaign/personnel/Person.java]: Unless you're using features specific to HashMap (that aren't in Map), you should declare these as Map.
From _Effective Java 2nd Edition, Item 52:_

> If appropriate interface types exist, then parameters, return values, and fields should all be declared using interface types. If you get into the habit of using interface types, your program will be much more flexible. It is entirely appropriate to refer to an object by a class if no appropriate interface exists.

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 2]
```diff
@@ -308,6 +312,23 @@
     //lets just go ahead and pass in the campaign - to hell with OOP
     private Campaign campaign;
 
+    // initializes the AtB ransom values
+    static {
+        MECHWARRIOR_AERO_RANSOM_VALUES = new HashMap();
```
Comment (dericpage) [inline MekHQ/src/mekhq/campaign/personnel/Person.java]: You're missing a diamond operator.  This should be 
`= HashMap<>()`

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 3]
```diff
@@ -308,6 +312,23 @@
...
+        MECHWARRIOR_AERO_RANSOM_VALUES.put(SkillType.EXP_ULTRA_GREEN, 5000);    // no official AtB rules for really inexperienced scrubs, but...
+        MECHWARRIOR_AERO_RANSOM_VALUES.put(SkillType.EXP_GREEN, 10000);
+        MECHWARRIOR_AERO_RANSOM_VALUES.put(SkillType.EXP_REGULAR, 25000);
+        MECHWARRIOR_AERO_RANSOM_VALUES.put(SkillType.EXP_VETERAN, 75000);
+        MECHWARRIOR_AERO_RANSOM_VALUES.put(SkillType.EXP_ELITE, 15000);
+        
+        OTHER_RANSOM_VALUES = new HashMap();
```
Comment (dericpage) [inline MekHQ/src/mekhq/campaign/personnel/Person.java]: Another missing diamond.

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 4]
```diff
@@ -3631,4 +3652,19 @@ public boolean hasChildren() {
         
         return hasKids;
     }
+    
+    // Returns the ransom value of this individual
+    // Useful for prisoner who you want to ransom or hand off to your employer in an AtB context
```
Comment (dericpage) [inline MekHQ/src/mekhq/campaign/personnel/Person.java]: JavaDoc comments should be preceeded with `/**` and end with `*/`
http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#format

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 5]
```diff
@@ -3631,4 +3652,19 @@ public boolean hasChildren() {
...
+    public int getRansomValue()
+    {
```
Comment (dericpage) [inline MekHQ/src/mekhq/campaign/personnel/Person.java]: Braces should be inline with their block declaration.

File: MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java [hunk 6]
```diff
@@ -682,6 +684,27 @@ public void actionPerformed(ActionEvent action) {
             case CMD_RECRUIT:
                 gui.getCampaign().changePrisonerStatus(selectedPerson, Person.PRISONER_NOT);
                 break;
+            case CMD_RANSOM:
+                // ask the user if they want to sell off their prisoners. If y
+                int total = 0;
+                for(Person person : people)
+                {
```
Comment (dericpage) [inline MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java]: Braces placement again.

File: MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java [hunk 7]
```diff
@@ -1223,6 +1246,13 @@ private void maybeShowPopup(MouseEvent e) {
                 popup.add(newMenuItem(resourceMap.getString("recruit.text"), CMD_RECRUIT, //$NON-NLS-1$
                         person.isBondsman() || person.isWillingToDefect()));
             }
+            
+            if(gui.getCampaign().getCampaignOptions().getUseAtB() &&
+               gui.getCampaign().getCampaignOptions().getUseAtBCapture() &&
+               StaticChecks.areAllPrisoners(selected))
+            {
```
Comment (dericpage) [inline MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java]: Braces

**Comment by NickAragua** [comment]
Made requested changes. The bracket thing is probably not going to go away very easily, I've been doing it the other way for about twenty years (and still do it that way at my job) so my pinkie automatically hits the enter key before a brace. I don't even know I'm doing it half the time.

**Pull request merged**